django_project/feti/static/js/scripts/views/occupation-view.js

```diff
@@ -19,18 +19,41 @@ define([
     var OccupationView = Backbone.View.extend({
         tagName: 'div',
         className: 'result-title result-title-occupation',
-        template: _.template('<div class="selected-indicator-right"><i class="fa fa-caret-right" aria-hidden="true"></i> </div><div class="title"><%- occupation %></div>'),
+        template: _.template('<div class="selected-indicator-right"><i class="fa fa-caret-right" aria-hidden="true"></i> </div><div class="title"><%- occupation %><p><span class="info-occupation-summary">details</span></p></div>'),
         detailTemplate: _.template(Detail),
         stepDetailTemplate: _.template(StepDetail),
         container: '#result-container-occupation',
         model: Occupation,
         occupationDetail: null,
         events: {
-            'click': 'clicked'
+            'click .info-occupation-summary': 'occupationDetailsClicked',
+            'click': 'occupationClicked'
         },
-        clicked: function () {
+        hide: function () {
+            this.model.hide();
+        },
+        occupationClicked: function () {
+            this.model.removeAllMarkers();
```

Comment:
maybe we could move this when the get_campus_provider call is success?
So the marker is not removed unless there is new marker from provider

Reply:
hi @meomancer , the problem is that not all occupations have provider (from the database). So if the marker is not removed unless there is new marker, this may create confusion isn't it? Since we may encounter the case where we click on occupation A but what's appear on the map is the markers from occupation B that we clicked before, because occupation A has no providers.
cmiiw

Reply:
@ann26, I think you are correct in this assumption.